File: app/src/main/java/se/accidis/fmfg/app/services/Preferences.java
package se.accidis.fmfg.app.services;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.HashSet;
import java.util.Set;

/**
 * Wrapper for Android shared preferences. Used to store application data between launches.
 */
public final class Preferences {
	public static final String PREFERENCES_FILE = "ThePreferences";
	private static final String TAG = Preferences.class.getSimpleName();
	private final SharedPreferences mPrefs;

	public Preferences(Context context) {
		mPrefs = context.getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE);
	}

	public String getDefaultAuthor() {
		return mPrefs.getString(Keys.DEFAULT_AUTHOR, "");
	}

	public Set<String> getFavoriteMaterials() {
		Set<String> set = mPrefs.getStringSet(Keys.FAVORITE_MATERIALS, null);
		return (set != null ? new HashSet<>(set) : new HashSet<String>());
	}

	public void setFavoriteMaterials(Set<String> materialsKeys) {
		mPrefs.edit().putStringSet(Keys.FAVORITE_MATERIALS, materialsKeys).apply();
	}

	public boolean shouldShowAuthorInDocument() {
		return mPrefs.getBoolean(Keys.SHOW_AUTHOR_IN_DOCUMENT, false);
	}

	public boolean shouldShowFbetInDocument() {
		return mPrefs.getBoolean(Keys.SHOW_FBET_IN_DOCUMENT, false);
	}

	public static class Keys {
		public static final String DEFAULT_AUTHOR = "DefaultAuthor";
		public static final String FAVORITE_MATERIALS = "FavoriteMaterials";
		public static final String SHOW_AUTHOR_IN_DOCUMENT = "ShowAuthorInDocument";
		public static final String SHOW_FBET_IN_DOCUMENT = "ShowFbetInDocument";

		private Keys() {
		}
	}
}

File: app/src/main/java/se/accidis/fmfg/app/services/Preferences.kt
package se.accidis.fmfg.app.services

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

/**
 * Wrapper for Android shared preferences. Used to store application data between launches.
 */
class Preferences(context: Context) {
    private val prefs: SharedPreferences =
        context.getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE)

    val defaultAuthor: String
        get() = prefs.getString(Keys.DEFAULT_AUTHOR, "")!!

    var favoriteMaterials: MutableSet<String>
        get() {
            val set = prefs.getStringSet(
                Keys.FAVORITE_MATERIALS,
                null
            )
            return if (set != null) HashSet(set) else HashSet()
        }
        set(materialsKeys) {
            prefs.edit {
                putStringSet(
                    Keys.FAVORITE_MATERIALS,
                    materialsKeys
                )
            }
        }

    fun shouldShowAuthorInDocument(): Boolean {
        return prefs.getBoolean(Keys.SHOW_AUTHOR_IN_DOCUMENT, false)
    }

    fun shouldShowFbetInDocument(): Boolean {
        return prefs.getBoolean(Keys.SHOW_FBET_IN_DOCUMENT, false)
    }

    object Keys {
        const val DEFAULT_AUTHOR: String = "DefaultAuthor"
        const val FAVORITE_MATERIALS: String = "FavoriteMaterials"
        const val SHOW_AUTHOR_IN_DOCUMENT: String = "ShowAuthorInDocument"
        const val SHOW_FBET_IN_DOCUMENT: String = "ShowFbetInDocument"
    }

    companion object {
        const val PREFERENCES_FILE: String = "ThePreferences"
    }
}
